packages/snaps-execution-environments/src/webview/ProxyMessageStream.ts

```diff
@@ -0,0 +1,109 @@
+import type { PostMessageEvent } from '@metamask/post-message-stream';
+import { BasePostMessageStream } from '@metamask/post-message-stream';
+import { isValidStreamMessage } from '@metamask/post-message-stream/dist/utils';
+import { base64ToBytes, bytesToString } from '@metamask/utils';
+
+type ProxyMessageStreamArgs = {
+  name: string;
+  target: string;
+  targetOrigin?: string;
+  targetWindow?: Window;
+};
+
+/**
+ * A {@link Window.postMessage} stream.
+ */
+
+export class ProxyMessageStream extends BasePostMessageStream {
```

Comment:
Perhaps this class should have another name. I'm just not sure what is should be called 😅

Reply:
That's a good one. From the previous architecture the proxy naming was already in place for what this class is doing, which is acting "in between". I can think about MiddlewareMessageStream or RouterMessageStream, but doesn't see like an important issue now and would rather keep for Proxy due old architecture compatibility.